src/common/database.c

```diff
@@ -2524,6 +2533,122 @@ gboolean dt_database_get_lock_acquired(const dt_database_t *db)
   return db->lock_acquired;
 }
 
+void _dt_database_vacuum(const struct dt_database_t *db)
+{
+  sqlite3_exec(db->handle, "VACUUM data", NULL, NULL, NULL);
+  sqlite3_exec(db->handle, "VACUUM data", NULL, NULL, NULL);
+  sqlite3_exec(db->handle, "ANALYZE main", NULL, NULL, NULL);
+  sqlite3_exec(db->handle, "ANALYZE main", NULL, NULL, NULL);
+}
+
+gboolean _ask_for_vacuum(const gboolean has_gui, guint64 size)
+{
+  if(!has_gui)
+  {
+    return 0;
+  }
+
+  char *size_info = g_format_size(size);
+
+  char *label_text = g_markup_printf_escaped(_("the database could use some maintenance\n"
+                                                 "\n"
+                                                 "there's <span style=\"italic\">%s</span> to be freed"
+                                                 "\n"
+                                                 "do you want to proceed now\n"),
+                                                 size_info);
+
+    gboolean shall_perform_maintenance =
+      dt_gui_show_standalone_yes_no_dialog(_("darktable - schema maintenance"), label_text,
+                                           _("later"), _("yes"));
+
+    g_free(label_text);
+    g_free(size_info);
+
+    return shall_perform_maintenance;
+}
+
+int _get_pragma_val(const struct dt_database_t *db, const char* pragma)
+{
+  gchar* query= g_strdup_printf("PRAGMA %s", pragma);
+  int val = -1;
+  sqlite3_stmt *stmt;
+  int rc = sqlite3_prepare_v2(db->handle, query,-1, &stmt, NULL);
+  if(rc == SQLITE_OK && sqlite3_step(stmt) == SQLITE_ROW)
+  {
+    val = sqlite3_column_int(stmt, 0);
+  }
+  sqlite3_finalize(stmt);
+  g_free(query);
+
+  return val;
+}
+
+void dt_database_maybe_vacuum(const struct dt_database_t *db, const gboolean has_gui, const gboolean closing_time)
+{
+  gboolean check_for_vacuum = FALSE;
+
+  char *config = dt_conf_get_string("database/maintenance_check");
+
+  if(config)
+  {
+    if(
```

Comment:
first condition should start after ( ; please be sure to use current code style.

Reply:
Damn. Where's current codestyle reference? I'd wish to setup my IDE...